### DISCUSSION TITLE
求助：插件突然失效 Plugin suddenly stop working

### DISCUSSION THREAD

**Discussion by FCJ1987** [DominikDoom/a1111-sd-webui-tagcomplete#225] [Q&A] [answered]
很抱歉困扰到大家，不过我的英语实在不是很好，与其自己拼写了也不知道语法准不准确，干脆还是写中文让翻译软件自动翻译好了。
---
接下来我说下我碰到的问题——在此之前，插件是完全可以正常使用的，直到我干了以下几件事：
我关闭了weibui的中文翻译插件，把webui调成了英文。
我在英文的webUI界面下，启用了edge浏览器的扩展程序monica，这是一款AI助理类的浏览器扩展程序。我尝试使用了它的网页翻译功能和AI拼写功能。
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/8d2af0cf-048f-4b55-aa6d-e6cb0a57eca5)
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/6a003e41-7bc9-4fb9-9915-1183bece212f)
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/a20e1599-5840-4cc2-92ba-9f6956d24d30)
这个时候，tagcomplete插件还能正常工作，直到我重启了一下webui，之后就再也没能正常工作了。这中间我没有调整过插件的设置界面，我相信插件的设置界面是没有什么重大的错误的
![img-2023 08 23-11 11 20](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/4601964e-e48f-4222-bb57-e30d50a89b62)

现在当我在prompt输入框内输入文字的时候，提示框完全不会弹出任何候选的提示词，是一种完全没有任何反应的无效状态。我尝试过关闭Monica，卸载和重新安装tagcomplete，重启电脑，都没有效果。

我的启动器上有这样一行报错，不知道是否和此有关：
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/75ce5ad7-753a-41d2-8778-68dd89d0ea7b)

---
有没有和我碰到类似情况的朋友，有什么有价值的思路或建议吗，谢谢了。
最后，非常感谢作者的工作，这个插件真的很棒！

**Comment by DominikDoom** [comment]
不用担心，现代翻译软件绝对足以让我理解您的意思。希望反过来也是如此😅。

启动器命令提示符中的信息并不是错误，它只是提示该扩展的一项功能，即支持基于外部列表的 Lora 关键字补全。它需要另一个扩展，所以信息中是这么说的。但如果没有它，插件也能正常工作，只是该功能仅限于你在 webui 额外网络菜单中自己添加的关键字。

关于实际问题，我试过monica和页面翻译，看看是否有冲突，但运行起来没有问题。只是翻译的格式有点乱，因为它是对原文的补充而不是替代。

在重新安装并完全重启后仍然无法正常工作，这肯定很奇怪，所以我肯定还有其他因素在起作用。
你能看看浏览器控制台吗？你可以用 CTRL+SHIFT+I 打开它，顶部应该有一个 "主控台" 选项卡。如果扩展的 javascript 部分有问题，就会显示在这里，而不是命令提示符窗口。

**Reply by FCJ1987** [reply]
作者大人您好，我查看了一下控制台，的确有一些和tagcomplete相关的错误提示，但是我没有编程知识，看不懂它。我把截图贴上来：
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/d5c1b392-6097-434c-ba50-223c21f43f39)

顺便一提，我刚刚还尝试了一下用之前安装用的压缩包，在其他位置解压安装了一份新的SD整合包，用那个新整合包登录可以正常使用tagcomplete，如你所说，那个正常使用的登录器上也有那个报错信息，但是并不影响插件使用。并且我把使用Monica的流程重复了一下，也没能复现问题，所以似乎不是浏览器的原因？希望之前的描述没有给你带来太大的误导。
当然，我上面所说“新”的SD整合包，实际上是一个比较“旧”的状态的，我的意思是，它是一个从比较早的压缩包里解压出来的“干净”的整合包，它还没来得及装很多其他的插件，不知道是不是有其他软件插件导致冲突。但是鉴于我几乎是眼睁睁的看着它从一次重启webui之后，就从之前的正常使用变成失效，这中间我几乎没有做什么修改设置或者安装插件的事，唯一做的事情就是设置了一下Monica并从webui上重启了浏览器。这也是我怀疑是因为Monica导致错误的原因，但是现在看起来似乎并不是这样。

已经很晚了，请您在其他方便的时候回复就好了，希望没有打扰到你的休息。

**Reply by DominikDoom** [reply]
嗯，这个错误似乎是在解析选项中的键映射 json 时出现的。也许是monica在翻译过程中在代码文本框中插入了一些内容，而你不小心应用了这些内容？

不过从你之前发送的截图来看是正常的，如果截图是最新的的话。
如果你想比较一下，这是原版：
```json
{
    "MoveUp": "ArrowUp",
    "MoveDown": "ArrowDown",
    "JumpUp": "PageUp",
    "JumpDown": "PageDown",
    "JumpToStart": "Home",
    "JumpToEnd": "End",
    "ChooseSelected": "Enter",
    "ChooseFirstOrSelected": "Tab",
    "Close": "Escape"
}
```

**Reply by FCJ1987** [reply]
是的，tagcomplete在webui设置界面中的截图是最新的，
这是对比，似乎并没有区别
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/e5e701e2-5fd9-4167-84d3-7523a04459e1)

**Reply by DominikDoom** [reply]
是的，看起来一切正常，不知道会是什么原因。你之前发布的错误信息中提到该选项的行号是源头，所以应该与此有关。

为确保万无一失，请尝试将 webui 根目录下的 "config.json" 文件暂时重命名为其他文件。该文件包含所有已更改/应用的选项，因此如果找不到，它会将所有内容重置为默认值。

如果没有任何变化，那么问题可能真的来自某些扩展或浏览器插件的交互。在这种情况下，在你提到的正在运行的辅助安装中一步步重新添加东西可能是找出问题原因的唯一办法。

**Reply by FCJ1987** [reply]
作者大人，我刚刚把config.json删除后再重启webui，现在tagcomplete已经能正常使用了。这意味着问题是出在webui的相关设置里吗，是否能在config.json里排除出具体的问题所在呢，就这样直接把config.json删除了的话，看启动器上的提示，其他插件似乎没有正常加载啊。

**Reply by DominikDoom** [reply, marked as answer]
那么一定是这个选项造成的。奇怪的是，从视觉上看一切正常，但文件中保存的内容可能不知何故被弄乱了。

如果你更改了任何设置，并希望保留这些设置而无需手动重新应用它们，请在原始 config.json 文件中搜索 "tac_keymap"，然后只删除这一行。也许还可以删除 "tac_colormap"，因为它使用的是同一种类型的 json 配置。如果只有这一处出错，效果应该是一样的。

**Reply by FCJ1987** [reply]
我把出问题的config.json和能够正常使用插件的副本安装包中的config.json放在一起对比了一下，tac_keymap这一行有差别，tac_colormap是一样的。
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/103525225/bda216b8-9148-4102-a1d6-502f25921730)
奇怪的是，从之前的截图里看不出异常，它们并没有体现在webui的设置界面上。我把异常的内容删除之后，重新启动webui就能正常使用了。
实在是太感谢了！话说您休息的可真晚啊，我在其他讨论帖里也看到过你在凌晨3点了还在回复帖子哈哈，辛苦了再次对以诚挚的感谢。

**Reply by DominikDoom** [reply]
不错！"\u00a0"是一个非中断空格符，所以在截图中看不到，因为它只是行尾的空白。

不用担心时间问题，哈哈。感谢时区的魔力，我的回复对我来说只是傍晚，而不是午夜之后。
